client/src/app/project-registration/autofill-project-form/autofill-project-form.component.ts

```diff
@@ -0,0 +1,80 @@
+import {Component, OnInit} from '@angular/core';
+import {FormControl, Validators} from '@angular/forms';
+import {ActivatedRoute, Router} from '@angular/router';
+import {AlertService} from '../../shared/services/alert.service';
+import {Identifier} from '../models/europe-pmc-search';
+import {Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
+import {IngestService} from '../../shared/services/ingest.service';
+
+@Component({
+  selector: 'app-doi-name-field',
+  templateUrl: './autofill-project-form.component.html',
+  styleUrls: ['./autofill-project-form.component.css']
+})
+
+
+export class AutofillProjectFormComponent implements OnInit {
+  publicationDoiCtrl: FormControl;
+
+  constructor(private route: ActivatedRoute,
+              private router: Router,
+              private ingestService: IngestService,
+              private alertService: AlertService
+  ) {
+  }
+
+  ngOnInit(): void {
+    this.publicationDoiCtrl = new FormControl('', Validators.required);
+  }
+
+  showError(control: FormControl): string {
+    if (control.touched && control.errors) {
+      const errors = control.errors;
+
+      if (errors['required']) {
+        return 'This field is required';
+      }
+    }
+  }
+
+  submitForm() {
+    if (this.publicationDoiCtrl.invalid) {
+      this.publicationDoiCtrl.markAsTouched();
```

Comment:
Can you get rid of the else? This is quite a lot of nesting. Just `return` from the first if.